src/Wait.hs

```diff
@@ -20,18 +22,22 @@ import System.FSNotify
 import Language.Haskell.Ghcid.Util
 
 
-data Waiter = WaiterPoll Seconds
-            | WaiterNotify WatchManager (MVar ()) (Var (Map.Map FilePath StopListening))
+data Waiter a
+  = WaiterPoll Seconds
+  | WaiterNotify WatchManager (MVar ()) (Var (Map.Map (FilePath, a) StopListening))
 
-withWaiterPoll :: Seconds -> (Waiter -> IO a) -> IO a
+withWaiterPoll :: Seconds -> (Waiter a -> IO b) -> IO b
 withWaiterPoll x f = f $ WaiterPoll x
 
-withWaiterNotify :: (Waiter -> IO a) -> IO a
+withWaiterNotify :: (Waiter a -> IO b) -> IO b
 withWaiterNotify f = withManagerConf defaultConfig{confDebounce=NoDebounce} $ \manager -> do
     mvar <- newEmptyMVar
     var <- newVar Map.empty
     f $ WaiterNotify manager mvar var
 
+firstM :: Monad m => (a -> m c) -> (a, b) -> m (c, b)
+firstM f (x, y) = fmap (, y) (f x)
```

Comment:
Wasn't sure about adding this one. It exists in other libraries, but I didn't want to add another dependency to `ghcid`.

Reply:
I just added it to extra-1.6.20 so if you adjust the dependency bounds you can use it from there.

Reply:
It's now in extra-1.6.20 so if you change the .cabal to require it, then you can use it from there.

Reply:
Done!